src/commands/apply.ts

```diff
@@ -54,18 +74,156 @@ export default class Apply extends Command {
+        mergeAll(5),
+        takeUntil(skippedStateNotifier),
+        catchError((f) => of(null)),
+        filter((f): f is ManifestFile => f !== null),
+        shareReplay()
+      );
+      try {
+        // step1: apply configmaps & secrets
+        await applyManifests(parsedFiles, ctx, 1, {
+          start: 'Applying configurations',
+        });
+        // step2: apply deployments and jobs
+        await applyManifests(parsedFiles, ctx, 2, {
+          start: 'Applying deployments',
+        });
+        // step3: apply other manifests
+        await applyManifests(parsedFiles, ctx, null, {
+          start: 'Applying other manifests',
+        });
+      } catch (error) {
+        // if error occurs, append msg to the running spinner
+        cli.action.stop('Error occured');
+        cli.action.start('Waiting until other manifests complete');
+        skipRemainingManifests();
+        // wait till all running tasks are completed or thrown
+        await parsedFiles
+          .pipe(
+            mergeMap((file) => file.manifests),
+            mergeMap((manifest) => manifest.waitTillCompletion())
+          )
+          .toPromise();
+        cli.action.stop();
       }
-      manifests.forEach((m) => {
-        this.log(`Applying ${m.kind} "${m.metadata?.name}"`);
+      const statusArr = await parsedFiles
+        .pipe(
+          mergeMap((file) => file.manifests.map((manifest) => manifest.state)),
+          toArray()
+        )
+        .toPromise();
+      const summary = new Map<ManifestState, number>();
+      statusArr.forEach((status) => {
+        summary.set(status, (summary.get(status) || 0) + 1);
       });
+      const manifestCount = statusArr.length;
+      const errCount = summary.get(ManifestState.ERROR) || 0;
+      const skippedCount = summary.get(ManifestState.SKIPPED) || 0;
+      // msg
+      const successCount =
+        (summary.get(ManifestState.UNKNOWN_SUCCESS_RESPONSE) || 0) +
+        (summary.get(ManifestState.FAILED_TO_WRITE_TO_FILE) || 0) +
+        (summary.get(ManifestState.COMPLETE) || 0);
+      if (manifestCount === 0) {
+        this.error(
+          `No valid manifests were found in the ${
+            isDir ? 'directory' : 'file'
+          }"${fileFolderPath}"`,
+          { exit: 1 }
+        );
+      } else if (manifestCount === successCount) {
+        cli.log(chalk.green(`Success`));
+      } else if (successCount === 0) {
+        cli.log(chalk.red(`Failure`));
+      } else {
+        cli.log(chalk.yellow(`Failed to apply some manifests.`));
+      }
+      // counts
+      cli.log('Manifests found: ', chalk.blue(manifestCount));
+      cli.log('Success count: ', chalk.blue(successCount));
+      cli.log(
+        'Errors occurred: ',
+        errCount === 0 ? chalk.blue(0) : chalk.red(errCount)
+      );
+      cli.log(
+        'Skipped manfests: ',
+        skippedCount === 0 ? chalk.blue(0) : chalk.yellow(skippedCount)
+      );
```

Comment:
I think we should also print out the applied filenames and their statuses (especially if they fail/get skipped).

kubectl prints something like this when you apply a bunch of resources -
Deployment x created
ConfigMap y configured
....

And the errors as well, if it's actually human-readable. (idk what rudder actually returns on error)

Reply:
done
![image](https://user-images.githubusercontent.com/12383824/103625557-28364780-4f61-11eb-8d87-0a497806268c.png)